src/main/java/org/wso2/carbon/inbound/custom/poll/MQPollingInbound.java

```
@@ -0,0 +1,221 @@
+
+            if (readableQueue == null) {
+                logger.info("Queue is not initialized properly");
+            } else {
+
+                MQMessage message = new MQMessage();
+                MQGetMessageOptions gmo = new MQGetMessageOptions();
+
+                if (config.getCorrelationID() != null && config.getMessageID() != null) {
+                    gmo.matchOptions = MQConstants.MQMO_MATCH_CORREL_ID + MQConstants.MQMO_MATCH_GROUP_ID;
+                } else if (config.getMessageID() != null) {
+                    gmo.matchOptions = MQConstants.MQMO_MATCH_MSG_ID;
+                } else if (config.getCorrelationID() != null) {
+                    gmo.matchOptions = MQConstants.MQMO_MATCH_CORREL_ID;
+                }
+
+                readableQueue.get(message, gmo);
+                logger.info("A message received");
+
+                MQMD md = new MQMD();
+                md.copyFrom(message);
+
+                message.getDataLength();
+                int strLen = message.getDataLength();
+                byte[] strData = new byte[strLen];
+                message.readFully(strData);
+
+                //crating new message context
+                msgCtx = this.createMessageContext();
+                msgCtx.setProperty("MessageId", new String(message.messageId));
+                msgCtx.setProperty("CorrelationID", new String(message.correlationId));
+                logger.info("Messsage data-" + new String(strData));
+
+                if (message.getStringProperty("ContentType") != null) {
+                    injectMessage(new String(strData), message.getStringProperty("ContentType"));
+                } else {
+                    injectMessage(new String(strData), MQConstant.DEFAULT_CONTENT_TYPE);
+                }
+                readableQueue.close();
+            }
+
+        } catch (Exception e) {
+            logger.info("Queue is empty");
```

This should be a error log and exception is dropped - logger.("Message", e);